src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

```diff
@@ -134,6 +163,19 @@ internal Task ConnectAsync(string host, int port)
             return ConnectAsync(ep);
         }
 
+        internal ValueTask ConnectAsync(string host, int port, CancellationToken cancellationToken)
```

Comment:
Can we reduce some of the duplication, such as by making the above method:
```C#
internal Task ConnectAsync(string host, int port) =>
     ConnectAsync(host, port, default).AsTask();
```
and the same anywhere else there's duplication?

Reply:
My hesitation here was to try to ensure we avoid an additional async state machine allocation for overloads that don't take a CancellationToken. I'll look at it again to see if I can remove some duplication.

Reply:
Ok

Reply:
I reworked this and address the sync cancellation issue above as well. Take a look if you have a chance.